server/src/middleware/auth-validator.ts

```diff
@@ -0,0 +1,23 @@
+import OAuth2Controller from "../controller/oauth2";
+import {User} from "../entity/user";
+
+export const authValidator = async (ctx, next) => {
+    const idToken = ctx.cookies.get("ti");
+    if (!idToken) {
+        ctx.state.isAuthenticated = false;
+    } else {
+        const ticket = await OAuth2Controller.verifyToken(idToken);
+        if (ticket) {
+            const user: User = await OAuth2Controller.getOrCreateUser(ticket.payload);
+            if (user) {
+                ctx.state.user = user;
+                ctx.state.isAuthenticated = true;
+            } else {
+                ctx.state.isAuthenticated = false;
```

Comment:
this should probably explicitly handle the case where a user can't be created such that the user is notified if this is the case

Reply:
same comment as above, added rudimentary http error status right now if they aren't logged-in or if an account wasn't created, but I don't want to invest time into this edge case.